src/t_string.c

```diff
@@ -99,7 +99,15 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
         return;
     }
 
-    setkey_flags |= (flags & OBJ_KEEPTTL) ? SETKEY_KEEPTTL : 0;
+    /* Set the flags for setKey() function */
+
+    /* In the following 2 cases we set the 'SETKEY_KEEPTTL' flag to avoid removing the expiration entry 
+     * from 'db->expires' :
+     * 1. When 'OBJ_KEEPTTL' is set in the flags passed in.
+     * 2. When 'expire' is not NULL. In this case, we will call setExpire() later and it will overwrite
+     * the expiration entry with a new TTL, so we don't need to remove the entry from db->expires now.
+     */
+    setkey_flags |= ((flags & OBJ_KEEPTTL) || expire) ? SETKEY_KEEPTTL : 0;
```

Comment:
This comment mostly describe what the code does. It is not necessary since we can read the code. I think a short comment, explaining why, is enough.

Idea: *When expire is not NULL, we avoid deleting so it can be updated later instead of first deleted and then created again.*

Reply:
follow your advice